### PR TITLE
Enable the transport and volumefs features

### PR DESCRIPTION
**Pull request by MikaelSmith**
Adds the ability to exec over SSH and to access filesystems using exec.

Signed-off-by: Michael Smith <michael.smith@puppet.com>

### CODE REVIEW

**Comment by MikaelSmith** [comment]
This is nearly ready, just want to setup some infra so Travis actually has something to test.

File: bolt.rb [hunk 1]
```diff
@@ -41,13 +59,38 @@ def list
...
 class Target < Wash::Entry
   label 'target'
-  state :connection_info
+  parent_of VOLUMEFS
+
+  def known_hosts(host_key_check)
+    return nil unless host_key_check == false
+
+    # Disable host key checking by redirecting known hosts to an empty file
+    Gem.win_platform? ? 'NUL' : '/dev/null'
```
Comment (ekinanp) [inline bolt.rb:125]: Wash doesn't work with Windows so why is this here? A comment would help.

Comment (MikaelSmith) [reply bolt.rb:125]: Was pretty much to avoid a future issue when it does.

**Comment by ekinanp** [comment]
Have some Qs/comments from looking at the repo's README.

* Re: plugin description, all I see is "presents groups and targets in the inventory, and enables SSH access to them". That sentence doesn't tell me why I'd want to or how I would use this plugin. Will that be added later, e.g. after a "boltwash" blog post?

* Do you think it's clear from Step (4) that the `wash.yaml` file should look something like:
```
external-plugins:
   script: <script>
bolt:
  <boltdir stuff here>
```
Asking b/c I don't know how likely it is that people could think that the `bolt` key is included somewhere else (like as a separate key in `external-plugins`). Ideally, it's probably best to make it as copy-pastable as possible. We could maybe try
```
external-plugins:
  script: <script>
# Uncomment this to get the inventory from a Boltdir instead
# of the default ~/.puppetlabs/bolt
#bolt:
# . dir: /path/to/boldir
```

* Not sure if mentioning the `bolt` executable should be a developer note, because right now I need to think a little bit to figure out the difference between the two.

**Comment by ekinanp** [comment]
I also wonder if we should update `wash` for a first-time user to do something like

```
Are you installing Wash just to use an external plugin (y/n)?
```

Reason is b/c I worry people who also use Docker/AWS/Kube will flock to Wash just to use Boltwash, then get angry b/c a specific core plugin isn't being configured. Maybe that's a non-issue.

Might even be worth doing something like
```
Are you installing Wash just to use an external plugin (y/n)?
Where's the plugin script located? <script_location>
Great! Your external plugin's been configured blah blah blah
```
where we can create a stub `wash.yaml` file for them containing the configured external plugin.

EDIT: Maybe we could (temporarily) do this just for Boltwash until we get enough users. So something like
```
Are you here just to use Boltwash (y/n)?
```
(We can shell out to gem install and gem contents for them)

**Comment by MikaelSmith** [comment]
For now I think it's still ok to have some install hoops to jump through.

I'll work on a better plugin description. It definitely overlaps with a blog post I plan to write.

**Comment by ekinanp** [comment]
I'm OK if you want to link to the blog post in the README to avoid repeating yourself (once the post is finished).

**Comment by MikaelSmith** [comment]
Updated with an intro.

**Review by ekinanp — COMMENTED** [review]
Great narrative! Some ideas:

* I think posting an `stree` layout and maybe `ls`'ing some things would help. Asking b/c "presents a view of Bolt targets organized by groups..." doesn't really paint a good picture of what that looks like in my head. It is also not clear to me that `webservers` is a group and `server1`, `server2` are targets -- that part is implicit later on (unless that's due to my Bolt ignorance). 

* Maybe add some descriptions to the entries (in the code)? Like "This is a group", "This is a target" is probably enough.

* Do you think it's worth adding a "Future improvements/ideas" section to the README? Then people would know that this thing is definitely something we intend to update with feedback. An initial improvement could be implementing `metadata` to return all the facts so you can use `find` to filter on them (if that'd be useful to a Bolt user).

**Comment by MikaelSmith** [comment]
Done. Working on how to add partial_metadata_schema, but I don't think that needs to block the PR.

File: bolt.rb [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
+  end
+end
+
 class Boltwash < Wash::Entry
   label 'bolt'
   is_singleton
   parent_of 'Group'
+  description <<~DESC
+    A plugin for Puppet Bolt's inventory. You can see target configuration in entry
+    metadata. You can SSH to targets and view their filesystems in 'fs' directories.
```
Comment (ekinanp) [inline bolt.rb:34]: You mean the `meta` command? Also, looks like `docs` doesn't mention `meta` unless there's at least one supported attribute. We should probably have a different message if the entry has no attributes, but has some partial metadata. Filed https://github.com/puppetlabs/wash/issues/697 to fix this.

Comment (MikaelSmith) [reply bolt.rb:34]: Yes, wasn't sure a better way to phrase it.

**Review by ekinanp — APPROVED** [review]

File: bolt.rb [hunk 3]
```diff
@@ -39,15 +69,88 @@ def list
   end
 end
 
+
 class Target < Wash::Entry
   label 'target'
-  state :connection_info
+  parent_of VOLUMEFS
+  description <<~DESC
+    This is a target. You can view target configuration with 'metadata', and SSH
```
Comment (ekinanp) [inline bolt.rb]: Ah I think this is where I meant to say "Did you mean the `meta` command"

Comment (MikaelSmith) [reply bolt.rb]: Ah yes. I'll update it.

**Review by ekinanp — APPROVED** [review]
